iotdb/src/main/java/org/apache/iotdb/db/query/control/OpenedFilePathsManager.java

```diff
@@ -89,22 +93,28 @@ public void removeUsedFilesForCurrentRequestThread() {
       long jobId = jobIdContainer.get();
       jobIdContainer.remove();
 
-      for (String filePath : filePathsMap.get(jobId)) {
-        FileReaderManager.getInstance().decreaseFileReaderReference(filePath);
+      for (String filePath : closedFilePathsMap.get(jobId)) {
+        FileReaderManager.getInstance().decreaseFileReaderReference(filePath, false);
       }
-      filePathsMap.remove(jobId);
+      closedFilePathsMap.remove(jobId);
+      for (String filePath : unclosedFilePathsMap.get(jobId)) {
+        FileReaderManager.getInstance().decreaseFileReaderReference(filePath, true);
+      }
+      unclosedFilePathsMap.remove(jobId);
     }
   }
 
   /**
    * Increase the usage reference of filePath of job id. Before the invoking of this method,
-   * <code>this.setJobIdForCurrentRequestThread</code> has been invoked, so <code>filePathsMap.get(jobId)</code> must
+   * <code>this.setJobIdForCurrentRequestThread</code> has been invoked, so <code>closedFilePathsMap.get(jobId)</code> must

```

Comment:
only `closedFilePathsMap` is not accurate, there may exist `unClosedFilePathsMap`

Reply:
fixed